File: app/src/main/java/org/movieos/feeder/model/Tagging.java

package org.movieos.feeder.model;

import com.google.gson.annotations.SerializedName;

import io.realm.RealmObject;
import io.realm.annotations.Index;
import io.realm.annotations.PrimaryKey;

public class Tagging extends RealmObject implements IntegerPrimaryKey {
    @PrimaryKey
    @SerializedName("id")
    int mId;

    @Index
    @SerializedName("feed_id")
    int mFeedId;

    @SerializedName("name")
    String mName;

    public Tagging() {
    }

    public int getId() {
        return mId;
    }

    public int getFeedId() {
        return mFeedId;
    }

    public String getName() {
        return mName;
    }
}

File: app/src/main/java/org/movieos/feeder/model/Tagging.kt
package org.movieos.feeder.model

import com.google.gson.annotations.SerializedName

import io.realm.RealmObject
import io.realm.annotations.Index
import io.realm.annotations.PrimaryKey

open class Tagging : RealmObject(), IntegerPrimaryKey {

    @PrimaryKey
    @SerializedName("id")
    override var id: Int = 0

    @Index
    @SerializedName("feed_id")
    var feedId: Int = 0
        internal set

    @SerializedName("name")
    var name: String? = null
        internal set
}
